internal/provider/mssql_permission_resource.go

```diff
@@ -0,0 +1,172 @@
+package provider
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/openaxon/terraform-provider-mssql/internal/core"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &MssqlPermissionResource{}
+var _ resource.ResourceWithImportState = &MssqlPermissionResource{}
+
+func NewMssqlPermissionResource() resource.Resource {
+	return &MssqlPermissionResource{}
+}
+
+type MssqlPermissionResource struct {
+	ctx core.ProviderData
+}
+
+type MssqlPermissionResourceModel struct {
+	Id              types.String `tfsdk:"id"`
+	Principal		types.String `tfsdk:"principal_id"`
+	Permission      types.String `tfsdk:"permission"`
+}
+
+func (r *MssqlPermissionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_permission"
+}
+
+func (r *MssqlPermissionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "DB permission resource",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "`<principal_id>/<permission>`.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+
+			},
+			"principal_id": schema.StringAttribute{
+				MarkdownDescription: "`<principal_id>`.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"permissions": schema.StringAttribute{
+				MarkdownDescription: "Name of database-level SQL permission. For full list of supported permissions, see [docs](https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-database-permissions-transact-sql?view=azuresqldb-current#remarks)",
+				Computed:            true,
```

Comment:
```suggestion
			"permission": schema.StringAttribute{
				MarkdownDescription: "Name of database-level SQL permission. For full list of supported permissions, see [docs](https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-database-permissions-transact-sql?view=azuresqldb-current#remarks)",
				Required:            true,
```

Either updating the MssqlPermissionResourceModel struct to ``Permissions []types.String `tfsdk:"permissions"` `` or this to ``permission``. The Computed will set this to be read-only data.